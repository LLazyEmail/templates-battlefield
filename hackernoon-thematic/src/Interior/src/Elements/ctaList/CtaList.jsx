import React from 'react';

import { a, li, p, div, strong } from './styles';

import { config } from './config';

import { Link } from 'react-emails-components-miscellaneous'; 

const CtaList = () => (
  <div dir="ltr" style={div}>
    <strong style={strong}>
      Ready to Claim Your Internet Name on Hacker Noon?
      </strong>
    <ol>
      <li dir="ltr" style={li}>
        <p dir="ltr" role="presentation" style={p}>
          <a href={config.link1} target="_blank" style={a}>Create a free account</a>.
          </p></li>
      <li dir="ltr" style={li}>
        <p dir="ltr" role="presentation" style={p}>
          <a href={config.link2} target="_blank" style={a}>Tell us what you want to read about</a>.
          </p></li>
      <li dir="ltr"
        style={li}>
        <p dir="ltr" role="presentation" style={p}>
          <a href={config.link3} target="_blank" style={a}>Submit your own tech stories</a>. 
          </p></li>
      <li dir="ltr" style={li}>
        <p dir="ltr" role="presentation" style={p}>
          <a href={config.link4} target="_blank" style={a}>Gain Access to Brand-As-Author</a>â„¢ Program. 
          </p></li>
      <li dir="ltr" style={li}>
        <p dir="ltr" role="presentation" style={p}>
          Get Your Tech Stories Featured on the <a href={config.link5} target="_blank" style={a}>Homepage of Hacker Noon</a>.
          </p>
      </li>
    </ol>
  </div>
);

export default CtaList;