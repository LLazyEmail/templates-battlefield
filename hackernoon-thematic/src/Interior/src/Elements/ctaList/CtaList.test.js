import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import CtaList from './index';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('CtaList/index component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CtaList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = create(<CtaList />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('CtaList should contain a list of elements', () => {
    const wrapper = shallow(<CtaList />);
    const list = wrapper.find('ol').exists();
    expect(list).toEqual(true);
  })

  it('ol in CtaList should contain at least 1 element', () => {
    const wrapper = shallow(<CtaList />);
    const list = wrapper.find('li').length > 1;
    expect(list).toEqual(true);
  })

  it('CtaList header', () => {
    const wrapper = shallow(<CtaList />);
    const list = wrapper.find('strong');
    expect(list.render().text()).toEqual("Ready to Claim Your Internet Name on Hacker Noon?");
  })

  it('CtaList links', () => {
    const wrapper = shallow(<CtaList />);
    const a = wrapper.find('a');
    expect(a.at(0).props()).toHaveProperty('href', 'https://hackernoon.com/signup?ref=noonifications.tech');
    expect(a.at(1).props()).toHaveProperty('href', 'https://app.hackernoon.com/subscriptions?ref=noonifications.tech');
    expect(a.at(2).props()).toHaveProperty('href', 'https://app.hackernoon.com/new?ref=noonifications.tech');
    expect(a.at(3).props()).toHaveProperty('href', 'https://sponsor.hackernoon.com/brand-as-author?ref=noonifications.tech');
    expect(a.at(4).props()).toHaveProperty('href', 'https://hackernoon.com/?ref=noonifications.tech');
  })

  it('CtaList links should be opened in a new window', () => {
    const wrapper = shallow(<CtaList />);
    const a = wrapper.find('a');
    a.forEach((_, index) => {
      expect(a.at(index).props()).toHaveProperty('target', '_blank');
    })
  })
});