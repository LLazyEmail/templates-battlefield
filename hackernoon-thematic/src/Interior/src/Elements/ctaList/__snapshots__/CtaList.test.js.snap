// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaList/index component renders correctly 1`] = `
<div
  dir="ltr"
  style={
    Object {
      "textAlign": "justify",
    }
  }
>
  <strong
    style={
      Object {
        "fontSize": "18px",
        "fontWeight": "bolder",
      }
    }
  >
    Ready to Claim Your Internet Name on Hacker Noon?
  </strong>
  <ol>
    <li
      dir="ltr"
      style={
        Object {
          "WebkitTextSizeAdjust": "100%",
          "msTextSizeAdjust": "100%",
          "msoLineHeightRule": "exactly",
        }
      }
    >
      <p
        dir="ltr"
        role="presentation"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "18px",
            "lineHeight": "125%",
            "margin": "10px 0",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "0",
            "textAlign": "left",
          }
        }
      >
        <a
          href="https://hackernoon.com/signup?ref=noonifications.tech"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "fontWeight": "bold",
              "msTextSizeAdjust": "100%",
              "msoLineHeightRule": "exactly",
              "textDecoration": "underline",
            }
          }
          target="_blank"
        >
          Create a free account
        </a>
        .
      </p>
    </li>
    <li
      dir="ltr"
      style={
        Object {
          "WebkitTextSizeAdjust": "100%",
          "msTextSizeAdjust": "100%",
          "msoLineHeightRule": "exactly",
        }
      }
    >
      <p
        dir="ltr"
        role="presentation"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "18px",
            "lineHeight": "125%",
            "margin": "10px 0",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "0",
            "textAlign": "left",
          }
        }
      >
        <a
          href="https://app.hackernoon.com/subscriptions?ref=noonifications.tech"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "fontWeight": "bold",
              "msTextSizeAdjust": "100%",
              "msoLineHeightRule": "exactly",
              "textDecoration": "underline",
            }
          }
          target="_blank"
        >
          Tell us what you want to read about
        </a>
        .
      </p>
    </li>
    <li
      dir="ltr"
      style={
        Object {
          "WebkitTextSizeAdjust": "100%",
          "msTextSizeAdjust": "100%",
          "msoLineHeightRule": "exactly",
        }
      }
    >
      <p
        dir="ltr"
        role="presentation"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "18px",
            "lineHeight": "125%",
            "margin": "10px 0",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "0",
            "textAlign": "left",
          }
        }
      >
        <a
          href="https://app.hackernoon.com/new?ref=noonifications.tech"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "fontWeight": "bold",
              "msTextSizeAdjust": "100%",
              "msoLineHeightRule": "exactly",
              "textDecoration": "underline",
            }
          }
          target="_blank"
        >
          Submit your own tech stories
        </a>
        . 
      </p>
    </li>
    <li
      dir="ltr"
      style={
        Object {
          "WebkitTextSizeAdjust": "100%",
          "msTextSizeAdjust": "100%",
          "msoLineHeightRule": "exactly",
        }
      }
    >
      <p
        dir="ltr"
        role="presentation"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "18px",
            "lineHeight": "125%",
            "margin": "10px 0",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "0",
            "textAlign": "left",
          }
        }
      >
        <a
          href="https://sponsor.hackernoon.com/brand-as-author?ref=noonifications.tech"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "fontWeight": "bold",
              "msTextSizeAdjust": "100%",
              "msoLineHeightRule": "exactly",
              "textDecoration": "underline",
            }
          }
          target="_blank"
        >
          Gain Access to Brand-As-Author
        </a>
        â„¢ Program. 
      </p>
    </li>
    <li
      dir="ltr"
      style={
        Object {
          "WebkitTextSizeAdjust": "100%",
          "msTextSizeAdjust": "100%",
          "msoLineHeightRule": "exactly",
        }
      }
    >
      <p
        dir="ltr"
        role="presentation"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontFamily": "\\"Source Sans Pro\\", \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "fontSize": "18px",
            "lineHeight": "125%",
            "margin": "10px 0",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "0",
            "textAlign": "left",
          }
        }
      >
        Get Your Tech Stories Featured on the 
        <a
          href="https://hackernoon.com/?ref=noonifications.tech"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "fontWeight": "bold",
              "msTextSizeAdjust": "100%",
              "msoLineHeightRule": "exactly",
              "textDecoration": "underline",
            }
          }
          target="_blank"
        >
          Homepage of Hacker Noon
        </a>
        .
      </p>
    </li>
  </ol>
</div>
`;
